README.md

```diff
@@ -1,178 +1,114 @@
 
-  Which means everything about the bot, not composer, not bot launcher. 
-  
-  Composer and bot launcher can be viewed as stateless applications, which only have a reference to bots. 
+####bots
+Each bot is defined by a ".bot" file, which includes all the references to the assets this bot would use. **note** this is not the shape of the .bot file currently used in SDK v4 of Bot Framework Emulator v4. The structure of the .bot file is subject to change. 
 
-* `Bot launcher` don't design with `composer` in mind. 
+## data flow
 
-  Bot launcher only depends and focus on `bots`. Nothing to do with composer. Even though composer will be able to run the launcher. 
+The bot's assets are the source of truth for the bot at any given time. The Application will hold dirty bot dialogs that will be lost until written into the bot's assets. The BotLauncher only references the state of the bot's assets and cannot read dirty bot assets held in the application.
 
-* `Composer` knows everything about `bots` and `bot launcher` 
-  
-  Composer needs to know how to start the launcher with property configuration to launcher specific bot. 
+<Needs Image>
 
 ## folder structure
 
-Then let's explore the structure of each component, and dive into some config files to show how this kind of intertion is supported by those files. 
-
-Here is a overview of the folder structure, with hightlighting on some important config files. 
+Here is a overview of a potential folder structure
 
     /- BotLauncher
       /- CSharp
-        /- appsettings.json (has a ref to xx.bot, otherwise which bot to run?)
-    /- Bots
-      /- SampleBot1
-        /- bot1.bot    (bot project file, maybe we can call it .botproject, all files should be referenced here)
+      /- Node
     /- Composer
       /- config.json   (the config to the composer)
-      /- launcher.json (composer use this config to start Bot Launcher)
+    ...somewhere else on disk
+    /-echo-bot
+      /.bot
 
+## a bot's .bot file
 
-It's very clearly divided in the top level, each component has it's own folder. 
+Here is an example of a .bot file intended to run on Node, with glob patterns for bot asset discovery.
 
-## .bot file
-The most important file in `Bots` folder is each .bot file, which defined a bot, containing all the references within this bot. 
 ```
-/Bots/SampleBot1/bot1.bot
-
 {
-    services: [
-        // luis
-        // qna
-    ],
-
-    files: [
-        ./**/*.lu
-        ./**/*.lg
-        ./**/*.dialog
-    ]
-    
-    entry: "./dialogs/start.dialog"
+  services: [
+    {
+      type: "luis",
+      endpoint: ${luis_endpoint},
+      app_id: ${luis_app_id},
+      token: ${luis_token} 
+    },
+    {
+      type: "qna",
+      endpoint: ${qna_endpoint},
+      app_id: ${qna_app_id},
+      token: ${qna_token} 
+    }
+  ],
 
-}
+  files: [
+    "./**/*.lu",
+    "./**/*.lg",
+    "./**/*.dialog"
+  ]
 
+  dialogEntry: "./dialogs/start.dialog",
+  start: "node index.js"
```

Comment:
it would be great if this command was provided by the Application and used by the BotLauncher to infer the run time and start the bot.

Reply:
I don't believe this command should be in .bot file, because it's a command to how to start the Launcher. you can have a .bot file, loaded by C# or node launcher, or whatever launcher. it's commands are different.

Reply:
what i'm suggesting is that instead of storing the launcher start command, we store the bots start command and generate the launcher command from it. we can infer the runtime by looking for `node` or `dotenv` in the command